Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,418 @@
+                this.get('container')
+                    .one('.ez-' + section + '-translation[data-languagecode="' + languageCode + '"]')
+                    .addClass(TRANSLATION_HIGHLIGHT_CLASS);
+            }
+        },
+
+        /**
+         * Sets the state for base translation checkbox
+         *
+         * @method _uiBaseTranslationState
+         * @protected
+         * @param {String} baseTranslation
+         */
+        _uiBaseTranslationState: function (baseTranslation) {
+            var c = this.get('container'),
+                baseCheckbox = c.one('.ez-base-translation-checkbox');
+
+            baseCheckbox.set('checked', baseTranslation);
+            if (baseTranslation) {
+                c.addClass(BASE_TRANSLATIONS_LIST_VISIBLE_CLASS);
+            } else {
+                c.removeClass(BASE_TRANSLATIONS_LIST_VISIBLE_CLASS);
+            }
+        },
+
+        /**
+         * Sets visibility of whole part containing base translations
+         *
+         * @method _uiSetBaseTranslationVisibility
+         * @protected
+         * @param {Boolean} canBaseTranslation
+         */
+        _uiSetBaseTranslationVisibility: function (canBaseTranslation) {
+            var c = this.get('container');
+
+            if (canBaseTranslation) {
+                c.addClass(BASE_TRANSLATION_AVAILABLE_CLASS);
+            } else {
+                c.removeClass(BASE_TRANSLATION_AVAILABLE_CLASS);
+            }
+        },
+
+        /**
+         * `selectedNewTranslationChange` event handler. It enables/disables the button
+         * depending on new translationselection
+         *
+         * @method _uiSetConfirmButtonState
+         * @protected
+         */
+        _uiSetConfirmButtonState: function () {
+            var confirmButton = this.get('container').one('.ez-languageselectionbox-confirm');
+
+            confirmButton.set('disabled', !this.get('selectedNewTranslation'));
+        },
+
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                title: this.get('title'),
+                newTranslations: this.get('newTranslations'),
+                existingTranslations: this.get('existingTranslations'),
+                canBaseTranslation: this.get('canBaseTranslation'),
+            }));
+
+            return this;
+        },
+
+        /**
+         * Cancel the languageSelectionBox by firing the `cancelLanguageSelect` event
+         *
+         * @protected
+         * @method _cancelLanguageSelection
+         */
+        _cancelLanguageSelection: function () {
+            this.fire('cancelLanguageSelect');
+        },
+
+        /**
+         * Confirm the languageSelectionBox's selection by firing the `languageSelected` event
+         *
+         * @protected
+         * @method _confirmLanguageSelection
+         */
+        _confirmLanguageSelection: function () {
+            this.fire('languageSelected');
```

Comment:
actually your comment makes me realize there's a deeper issue in your patch. Basically when the user chooses a language first the `confirm` event is fired and then the `languageSelected` event is fired. The second one is used to hide the side view while the first one is the one you use with the confirm handler. But there should be only one event (`languageSelected`). This one should be triggered when the user uses the confirm button and its event facade should contain the selected language code and the base language code if any. The `confirmHandler` should then be renamed to `languageSelectedHandler`.
The exact same thing is valid for the case where the user uses the cancel button.

I guess you kind of mixed the confirm box event system with the universal discovery event system while the language selection box is actually close to the universal discovery.